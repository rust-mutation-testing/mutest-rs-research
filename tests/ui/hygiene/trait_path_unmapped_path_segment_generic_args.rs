//@ build
//@ stderr
//@ mutest-flags: --depth=1 --call-graph-depth-limit=1

//! The following paths get normalized into paths that point to their trait items,
//! which means that the generic arguments in the original path segments will be discarded.
//! This is fine however, since we then add the qualified self type
//! from the type system, e.g. `<Vec<u8> as Default>::default`,
//! ensuring that resolution does not fail or change.
//!
//! This test is here to document this behavior and ensure that
//! paths like these will remain resolvable.

#![feature(decl_macro)]

macro m() {
    let _ = Vec::<u8>::default();
    let _ = std::collections::HashSet::<i32>::default();
}

#[test]
fn test() {
    m!();
}
