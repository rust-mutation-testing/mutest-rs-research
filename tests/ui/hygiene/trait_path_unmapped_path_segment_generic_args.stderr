warning: incomplete call graph due to explicit depth limit
  |
  = note: call graph depth limit is set to 1
  = note: ignoring 3 callers and their callees

warning: 1 warning emitted

